### PR TITLE
Change markdown template for iam resource to show required field on argument reference

### PR DESCRIPTION
**Pull request by hao-nan-li**
fixes [15909](https://github.com/hashicorp/terraform-provider-google/issues/15909)

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by hao-nan-li** [comment]
For [this particular ](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/gke_hub_membership_iam#google_gke_hub_membership_iam_policy), I don't see `location` marked as `required`. 

I checked on https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/gke_hub_feature_iam, where `location` is required, that field has a `required` tag on it. https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/gkehub2/Feature.yaml#L94

Is there anything I missed?

File: mmv1/templates/terraform/resource_iam.html.markdown.erb [hunk 1]
```diff
@@ -217,6 +217,9 @@ resource "<%= resource_ns_iam -%>_member" "member" {
 
 The following arguments are supported:
 
+<% object.root_properties.select(&:required).each do |prop| -%>
```
Comment (rileykarson) [inline mmv1/templates/terraform/resource_iam.html.markdown.erb:220]: Note that this seems to generate bad entries for `name` and `location` in some cases

Comment (rileykarson) [reply mmv1/templates/terraform/resource_iam.html.markdown.erb:220]: And invalid ones like `replica_zones` for region disk

Comment (hao-nan-li) [reply mmv1/templates/terraform/resource_iam.html.markdown.erb:220]: Noted, for some iam files this looks like generated duplicated entries for `name` and `location`

**Comment by hao-nan-li** [comment]
Not hitting the expected result after the change, will take another look tomorrow.

**Comment by melinath** [comment]
> For [this particular ](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/gke_hub_membership_iam#google_gke_hub_membership_iam_policy), Looks like `location` is not marked as `required`.
> 
> I checked on https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/gke_hub_feature_iam, where `location` is required, that field has a `required` tag on it. https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/gkehub2/Feature.yaml#L94
> 
> Is there anything I missed?

IAM location fields are never required - they're Optional (+ Computed) - see https://github.com/GoogleCloudPlatform/magic-modules/blob/4146c391a677e649b213fe5e028bcec393ac3c6a/mmv1/templates/terraform/iam_policy.go.erb#L45-L47 That seems like a mistake in the docs.

**Review by melinath — COMMENTED** [review]
to get the correct set of fields to document, I'd recommend duplicating the logic used to generate the actual IAM resource schemas if possible, since that's what we want to match here: https://github.com/GoogleCloudPlatform/magic-modules/blob/4146c391a677e649b213fe5e028bcec393ac3c6a/mmv1/templates/terraform/iam_policy.go.erb#L41

**Comment by hao-nan-li** [comment]
> to get the correct set of fields to document, I'd recommend duplicating the logic used to generate the actual IAM resource schemas if possible, since that's what we want to match here:
> 
> https://github.com/GoogleCloudPlatform/magic-modules/blob/4146c391a677e649b213fe5e028bcec393ac3c6a/mmv1/templates/terraform/iam_policy.go.erb#L41

Thanks for suggesting, will take a look into it.

**Pull request closed without merging**